onmt/inputters/image_dataset.py

```diff
@@ -3,60 +3,19 @@
 import codecs
 import os
 
-from onmt.inputters.dataset_base import DatasetBase
-
-
-class ImageDataset(DatasetBase):
-    """
-    Build `Example` objects, `Field` objects, and filter_pred function
-    from image corpus.
-
-    Args:
-        fields (dict): a dictionary of `torchtext.data.Field`.
-        src_examples_iter (dict iter): preprocessed source example
-            dictionary iterator.
-        tgt_examples_iter (dict iter): preprocessed target example
-            dictionary iterator.
-        num_src_feats (int): number of source side features.
-        num_tgt_feats (int): number of target side features.
-        tgt_seq_length (int): maximum target sequence length.
-        use_filter_pred (bool): use a custom filter predicate to filter
-            out examples?
-    """
+from onmt.inputters.dataset_base import AudioVisualDataset
+
+
+class ImageDataset(AudioVisualDataset):
+    data_type = 'img'  # get rid of this class attribute asap
```

Comment:
Are you sure we can't get rid of it already ?

Reply:
In the newest batch of commits, I got rid of the need for it at training time, but there's more places to look at in the translator. The way the translator currently interacts with the data is generally not very pretty. I think incremental simplification will make it clearer how to remove it.